implementations/rust/ockam/ockam_command/src/service/config.rs

```
@@ -44,6 +50,64 @@ pub struct SecureChannelListenerConfig {
     pub(crate) identity: Option<String>,
 }
 
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, Default)]
+#[serde(rename_all = "kebab-case")]
+pub enum ControlApiNodeResolution {
+    #[default]
+    Relay,
+    DirectConnection,
+}
+
+fn default_control_api_bind_address() -> SocketAddr {
+    SocketAddr::from(([0, 0, 0, 0], 4080))
+}
+
+fn default_frontend_policy() -> PolicyExpression {
+    BooleanExpression(BooleanExpr::from_str("node_control_api_frontend").unwrap())
+}
+
+fn default_backend_policy() -> PolicyExpression {
+    BooleanExpression(BooleanExpr::from_str("node_control_api_backend").unwrap())
+}
+
+fn default_node_port() -> u16 {
+    4100
```

You could use the `Port` struct here. It has a `TryExplicitOrRandom` variant that can be useful to fallback to a random port if the default port is already in use.

---

Sorry, it isn't clear with the current naming, but the port is not a bind one but a connection port when connecting to other nodes. When the node resolution is set to 'connect', requests only provide the node name but not the port, and hence there is the need for a common port.
 I've renamed the port and the default to `connection_node_port`.

---

Ok nice, that makes things clearer, thanks!